submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/submitter/SubmissionSubmitter.kt

```diff
@@ -41,15 +44,23 @@ open class SubmissionSubmitter(
     private val accNoService: AccNoService,
     private val parentInfoService: ParentInfoService,
     private val projectInfoService: ProjectInfoService,
-    private val context: PersistenceContext
+    private val context: PersistenceContext,
+    private val queryService: SubmissionQueryService
 ) {
     @Transactional(isolation = Isolation.READ_UNCOMMITTED)
     open fun submit(request: SubmissionRequest): Submission {
         val user = request.user.asUser()
-        val extSubmission = process(request.submission, request.user.asUser(), request.files, request.method)
+        val sources = sources(request.submission.accNo, request.sources)
+        val extSubmission = process(request.submission, request.user.asUser(), sources, request.method)
         return context.saveSubmission(extSubmission, user.email, user.id).toSimpleSubmission()
     }
 
+    private fun sources(accNo: String, sources: FilesSource): FilesSource =
+        when (val submissionFolder = queryService.getExistingFolder(accNo)) {
+            null -> sources
+            else -> ComposedFileSource(listOf(sources, PathFilesSource(submissionFolder.toPath())))
```

Comment:
Maybe, instead of creating a `ComposedFileSource` before in `SourceGenerator` and then using it here to add the submission folder itself, we could just get the files that the user attaches to the request (if any) and then, use the SourceGenerator here and generate all at once.

It's just that's a bit confusing to do one part in one place and the rest here and besides, the files aren't actually required until the persistence process where they're actually used.